Tokens.API/Program.cs

```diff
@@ -41,3 +47,41 @@ static void AddAzureAppConfiguration(WebHostBuilderContext hostingContext, IConf
                 .Select("*", "Tokens");
         });
 }
+static void AddLoggingConfiguration(IConfigurationRoot envConfiguration)
+{
+    var environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
+    var service = Assembly.GetExecutingAssembly().GetName().Name.ToLower().Replace(".", "-");
+    var configuration = new ConfigurationBuilder()
+    .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
+    .AddJsonFile(
+        $"appsettings.{environment}.json",
+        optional: true)
+    .Build();
+
+    var elasticStackConfiguration = envConfiguration.GetElasticStackConfiguration();
+
+    if (elasticStackConfiguration.Enabled && !string.IsNullOrEmpty(elasticStackConfiguration.Host))
+    {
+
+        Log.Logger = new LoggerConfiguration()
+           .Enrich.FromLogContext()
+           .WriteTo.Debug()
+           .WriteTo.Console()
+           .WriteTo.Http($"http://logstash:{elasticStackConfiguration.Host}", null)
+           .Enrich.WithProperty("Environment", environment)
+           .Enrich.WithProperty("Service", service)
+           .ReadFrom.Configuration(configuration)
+           .CreateLogger();
+    }
+    else
+    {
+        Log.Logger = new LoggerConfiguration()
+           .Enrich.FromLogContext()
+           .WriteTo.Debug()
+           .WriteTo.Console()
+           .Enrich.WithProperty("Environment", environment)
+           .Enrich.WithProperty("Service", service)
+           .ReadFrom.Configuration(configuration)
+           .CreateLogger();
+    }
```

Comment:
The following should do the same, with less code duplication:

```suggestion
    var loggerConfiguration =  new LoggerConfiguration()
           .Enrich.FromLogContext()
           .WriteTo.Debug()
           .WriteTo.Console()
           .Enrich.WithProperty("Environment", environment)
           .Enrich.WithProperty("Service", service)
           .ReadFrom.Configuration(configuration);
           
    
    if (elasticStackConfiguration.Enabled && !string.IsNullOrEmpty(elasticStackConfiguration.Host))
    {
           loggerConfiguration = loggerConfiguration.WriteTo.Http($"http://logstash:{elasticStackConfiguration.Host}", null);
    }
    
    Log.Logger = loggerConfiguration.CreateLogger();
```